images/virtualization-artifact/pkg/controller/vmop/internal/util.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+
+func IsFinalState(vmop *virtv2.VirtualMachineOperation) bool {
+	if vmop == nil {
+		return false
+	}
+
+	if vmop.Status.Phase == virtv2.VMOPPhaseCompleted || vmop.Status.Phase == virtv2.VMOPPhaseFailed {
+		return true
+	}
+
+	return false
+}
```

Comment:
consider adding this function as a method to a resource VMOP.

Reply:
Moved to vmop service